src/main/java/roboy/context/ObservableValue.java

```diff
@@ -0,0 +1,24 @@
+package roboy.context;
+
+import java.util.Observable;
+
+/**
+ * A Value that supports adding Observers. These will be notified whenever a new value is added.
+ * @param <V>
+ */
+public class ObservableValue<V> extends Observable implements AbstractValue<V> {
+    private V value = null;
+
+    @Override
+    public V getValue() {
+        return value;
+    }
+
+    @Override
+    public synchronized void updateValue(V value) {
+        this.value = value;
+        setChanged();
+        notifyObservers();
+        clearChanged();
```

Comment:
``clearChanges()`` is not needed here as ``notifyObservers()`` calls it automatically. But it doesn't hurt either. 

https://docs.oracle.com/javase/7/docs/api/java/util/Observable.html